vendor/tests/test_processor.py

```diff
@@ -973,12 +1009,14 @@ def test_cancel_subscription_success(self):
             gateway_id=active_subscriptions[0].id.pyval,
             profile=CustomerProfile.objects.get(pk=1),
         )
+
         if active_subscriptions:
-            self.processor.subscription_cancel(dummy_subscription)
-            self.assertTrue(self.processor.transaction_submitted)
+            with self.assertRaises(Exception):
+                self.processor.subscription_cancel(dummy_subscription)
```

Comment:
just curious, why does this cancellation fail, as opposed to the previous test?

Reply:
If you take a look at the `_success` case you can see that I need to create a settled payment and in the `_fail` case I dont. You cant cancel a subscription that has no settled payments.

Reply:
Take a look at this function `subscription_cancel` in `processors/base.py`